app/controllers/patients_controller.rb

```diff
@@ -265,6 +265,52 @@ def update_hoh
     current_user_patients.where(id: patients_to_update).update_all(responder_id: new_hoh_id)
   end
 
+  def bulk_update_status
+    redirect_to(root_url) && return unless current_user.can_edit_patient?
+
+    patient_ids = params.permit(ids: [])[:ids]
+
+    # If apply to group, find dependents ids and add to id array before user accessor for validation of access
+    if params.permit(:apply_to_group)[:apply_to_group]
+      dependent_ids = Patient.where(responder_id: params.permit(ids: [])[:ids]).where.not(id: params.permit(ids: [])[:ids]).select('id')
+      patient_ids = patient_ids.union(dependent_ids)
+    end
+
+    patients = current_user.get_patients(patient_ids)
+
+    patients.where.not(monitoring: params.require(:patient).permit(:monitoring)[:monitoring]).where(monitoring: true).update_all(closed_at: DateTime.now)
+    patients.update_all(params.permit(:monitoring, :monitoring_reason, :monitoring_plan,
+                                      :exposure_risk_assessment, :public_health_action,
+                                      :isolation, :pause_notifications, :symptom_onset,
+                                      :case_status).to_h)
```

Comment:
This list of fields is missing `assigned_user` because it was added between the time the branch was checked out and the time of rebase.

It would also be more ideal for this to be refactored in a way (maybe in the future if there's no time now since I know this implementation is slightly different because it's doing multiple patients at once) to be integrated with this method for consistency so that every time we add a field or something to this list, we won't have to worry about adding it to 2 places (we've encountered this problem in the past before): https://github.com/SaraAlert/SaraAlert/blob/4a4c9f258648bd64b95ea45f59d28d4321e25aad/app/controllers/patients_controller.rb#L290

Reply:
I'm wondering about this and if we wanted to dry it out and consolidate to one spot, if it would be ideal to change the update_fields/history logic to use that of a bulk edit and then have single entry updates be a alias to the bulk where it wraps it just performs the bulk style entries for a single entry. The pro is DRYed out code, the con is additional lose of validations.

Reply:
I definitely like the idea of abstracting the logic into one place but at the same time I think the validation is valuable. @holmesie thoughts?

Reply:
Of course there is the option of the reverse as well, where bulk edits loop over individual calls to update_fields. Pros: Full validation, DRYed out code. Cons: Bulk edits time and db queries go from O(1) to O(n).

Reply:
maybe another intermediate alternative is that we can add our own code to do some or all of the validation checks in the controller before calling update_all

Reply:
My vote would be to update each at once for now since it seems to be the safer way and we can only update the amount of patients on a page so its not going to be terrible...

Reply:
I also wouldn't mind switching to the generic `update_fields` that would save a ton of logic in this method.